src/screens/ExchangeModal.js

```diff
@@ -266,15 +271,17 @@ export default function ExchangeModal({
 
   // Update gas limit
   useEffect(() => {
-    updateGasLimit();
-  }, [updateGasLimit]);
+    if (!isEmpty(gasPrices)) {
+      updateGasLimit();
+    }
+  }, [gasPrices, updateGasLimit]);
 
   // Set default gas limit
   useEffect(() => {
-    setTimeout(() => {
```

Comment:
Nit: can we just move this `useEffect` up before the `// Update gas limit` useEffect? Since technically they are run in order, and it makes sense for the default one to be tried first.